### PR TITLE
feat: improve visibility of the follow cta on user profiles

### LINKED ISSUE #1727: Feature: Improve Visibility of the Follow CTA on User Profiles
### Type of feature

🧑‍💻 Refactor

### Current behavior

Currently you have to click on the three dots menu to see the Follow/Unfollow CTA. This was not obvious when I went to follow a new user on the platform.

image

### Suggested solution

Have the Follow button placed beside the Share button.

### Additional context

Slack Discussion: https://opensaucedworkspace.slack.com/archives/C035RQ8ANBW/p1695223459466669

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shavidze**
## Description
Let's see what was before and what is after: 
[__See what was before__](https://github.com/open-sauced/app/issues/1727)
__After__:
 __LAST UPDATE__
@isabensusan 

https://screenapp.io/app/#/shared/2d292355-6af8-498f-b439-784f72465525

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1727

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Hey @shavidze, Thanks for taking this on 👏. Would you add a before/after screenshot to speed up the review process?

**Comment by shavidze** [comment]
> Hey @shavidze, Thanks for taking this on 👏. Would you add a before/after screenshot to speed up the review process?

Updated in description

**Comment by nickytonline** [comment]
This looks good on a large screen. Ensure it still looks good on mobile.

Provide screenshots for smaller screens. The three dots menu may still be required for mobile views.

**Comment by shavidze** [comment]
> Thanks for the pull request @shavidze!
> 
> This looks good on a large screen. Ensure it still looks good on mobile.
> 
> Provide screenshots for smaller screens. The three dots menu may still be required for mobile views.

Good note, I will return the three dots on mobile

**Comment by shavidze** [comment]
> Thanks for the pull request @shavidze!
> 
> This looks good on a large screen. Ensure it still looks good on mobile.
> 
> Provide screenshots for smaller screens. The three dots menu may still be required for mobile views.

__updated__ in description

**Comment by isabensusan** [comment]
hey @shavidze ! thanks for the good work. Here's a couple of design comments:
1. Let's keep only 2 visible CTA's, "Follow" (make it the primary CTA = make it orange)  and "Get Card" (make it white). Let's add the three dots dropdown to desktop too and put the other options there ("Collaborate" and "Share"). 

Desktop should look sth like this, and if the user presses on the three dots they should be able to see "Collaborate" and "Share" in the dropdown.
image

For mobile, keep the same structure as with desktop, but to make the "follow" and "get card" CTA's smaller, show only the icons.


2. Style fix: the CTA you've introduced looks different than the ones that were there before. Can we try and re-use the same component as the other CTA's (specifically the share CTA) on the page? 


Thanks!

**Comment by shavidze** [comment]
@isabensusan 
Thanks for your comment.

About the second note, I didn't change the CTA's styles I just removed dots and got the icons on the screen.

**Comment by OgDev-01** [comment]
Feedback after the recent change:

The follow  button on the desktop view should be an actual `Button` component with a variant prop of `primary`.
here is the button for reference.
https://github.com/open-sauced/insights/blob/34dead8f963f6f89d6a16ee3e3f447e266373d17/components/atoms/Button/button.tsx

We also should have the `+user` icon in the following CTA on desktop
Screenshot 2023-09-26 at 10 44 02

like this
image

**Comment by shavidze** [comment]
> Feedback after the recent change:
> 
> The follow button on the desktop view should be an actual `Button` component with a variant prop of `primary`. here is the button for reference. https://github.com/open-sauced/insights/blob/34dead8f963f6f89d6a16ee3e3f447e266373d17/components/atoms/Button/button.tsx
> 
> We also should have the `+user` icon in the following CTA on desktop Screenshot 2023-09-26 at 10 44 02
> 
> like this image

@OgDev-01 Okay, That should not be a problem, you see that the button for the visitor looks like this, I use this primary button,
My question is what will be the design for the signed-in user, when she/he presses this follow button?

**Comment by OgDev-01** [comment]
>if the user is signed in and presses the Follow, what would this scenario look like? I mean how will this Following/Unfollow button look especially on mobile?

Mobile should also be orange, and the button should show `following` if I follow the user(with a hover effect that changes the text in the button to `unfollow`) or `follow` if I don't.
Screenshot 2023-09-26 at 10 53 48

clicking on the follow button should update the state based on the following condition

**Comment by OgDev-01** [comment]
This is what happens when a signed-in user clicks on the button if this is what you meant by
>" what will be the design for the signed-in user, when she/he presses this follow button?"

https://github.com/open-sauced/app/assets/62995161/32763c61-09d3-43de-8aff-1b0ac38b08e4

Additionally, if you meant what it should look like on the signed-in user profile, then we should have just the share and get card button
Screenshot 2023-09-26 at 11 04 10

**Comment by OgDev-01** [comment]
Let me know if I answered your question @shavidze 👍, and also feel free to ask if you have any other questions 🤝

**Comment by shavidze** [comment]
> This is what happens when a signed-in user clicks on the button if this is what you meant by
> 
> > " what will be the design for the signed-in user, when she/he presses this follow button?"
> 
>  Screen.Recording.2023-09-26.at.10.59.40.mov 
> 
> Additionally, if you meant what it should look like on the signed-in user profile, then we should have just the share and get card button Screenshot 2023-09-26 at 11 04 10

So, just changing the button to primary, without any hovering effect for the text

**Comment by shavidze** [comment]
> The hover on the following/follow button should still change the text, but the button background should remain orange

👍😀 yes, text should be changed
Okay, I will make changes soon

**Comment by shavidze** [comment]
@OgDev-01 **Updated Video In Description**

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 1]
```diff
@@ -155,96 +156,116 @@ const ContributorProfileHeader = ({
...
+                </>
+              )}
 
-                {!isOwner && (
-                  <DropdownMenuTrigger title="More options" className="p-2 mr-3 bg-white rounded-full cursor-pointer ">
-                    <TfiMoreAlt size={20} className="" />
-                  </DropdownMenuTrigger>
-                )}
-              </div>
+              <Button className="sm:hidden" variant="primary" href={cardPageUrl(username!)}>
```
Comment (OgDev-01) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:198]:
```suggestion
              <Button className="sm:hidden" variant="text" href={cardPageUrl(username!)}>
```

**Review by OgDev-01 — APPROVED** [review]
minor updates and this is good 🍕🍕🍕

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx (hunk 1 above)
Comment (shavidze) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:198]: > Do you want to change the Get Card Button variant to text for mobile?

Comment (OgDev-01) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:198]: yes... so it is with and the icon is orange

Comment (shavidze) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx]: Done 👌

Comment (OgDev-01) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx]: sorry for the back and forth 😄, can you please add a `bg-white` class name to that button to make it white?

Comment (shavidze) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx]: sure, done 🚀

**Comment by isabensusan** [comment]
lgtm! thanks for the great work and for iterating on it too @shavidze !! 🥇  🥇

**Comment by nickytonline** [comment]
> For mobile, keep the same structure as with desktop, but to make the "follow" and "get card" CTA's smaller, show only the icons.

As @isabensusan mentioned, we want the follow button to be an icon only on smaller screens. Currently it's still text.

Here's an example of the iPhone screen size. It fits, so maybe we're good with the follow CTA being textual even on mobile Isa?

CleanShot 2023-09-27 at 15 38 03

Maybe not related to this PR, maybe it's our buttons in general, but do we want text centered or not?

**Comment by brandonroberts** [comment]
@nickytonline I think the text should be centered, and we can follow-up with the button

**Comment by shavidze** [comment]
> @nickytonline I think the text should be centered, and we can follow-up with the button

Thanks! Done.

**Review by brandonroberts — APPROVED** [review]
Nice work!

**Comment by nickytonline** [comment]
Thanks for this @shavidze! 🥳

**Pull request merged**